vision/objectdetection/color_detector.go

```diff
@@ -30,6 +30,10 @@ func NewColorDetector(cfg *ColorDetectorConfig) (Detector, error) {
 		return nil, err
 	}
 	hue, s, v := col.HsvNormal()
+	if s == 0 { // color detector cannot detect black/white/grayscale
+		return nil, errors.New("the chosen color to detect has a saturation of 0. " +
+			"The color detector cannot detect black, white or grayscale colors.")
+	}
 	tol := cfg.HueTolerance
 	sat := cfg.SaturationCutoff
 	if sat == 0 {
```

Comment:
Would be nice to add a comment to explain why the default values are these i.e. 0.2 for sat, 0.3 for value cutoff